resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/local/unix/UnixFileNatives.java

```diff
@@ -58,6 +71,22 @@ private static boolean isLibraryPresent() {
 		return entries != null && entries.hasMoreElements();
 	}
 
+	private static boolean hasBundle(String pattern) {
+		Pattern compile = Pattern.compile(pattern);
+		Bundle bundle = FrameworkUtil.getBundle(UnixFileNatives.class);
+		if (bundle != null) {
```

Comment:
See above, can we simplify the nesting by inverting the conditions?

Reply:
I don't see what should be "inverted" here, can you please explain in more detail?

Reply:
naively, I would write 
```java
if (bundle == null)
  return false;
BundleContext bundleContext = bundle.getBundleContext();
if (bundleContext == null)
  return false;
for (Bundle other : bundleContext.getBundles()) {
  if (compile.matcher(other.getSymbolicName()).matches()) {
    return true;
  }
}
return false;
```